File: C++ Kasus/src/Penjual.cpp
#include "myclass/Penjual.h"
#include <iostream>

std::string Penjual::user()
{
    return "ini Penjual";
}

Penjual::Penjual(std::string nama, std::string gender, int umur, int lamaKerjaPerTahun) : User(nama, gender, umur)
{
    this->lamaKerjaPerTahun = lamaKerjaPerTahun;
}

void Penjual::infoPenjual()
{
    User::infoUser();
    std::cout << " tahun, Lama Kerja : " << lamaKerjaPerTahun << " Tahun" << std::endl;
}

void Penjual::setLamaKerjaPerTahun(int waktu)
{
    this->lamaKerjaPerTahun = waktu;
}

int Penjual::getLamaKerjaPerTahun()
{
    return this->lamaKerjaPerTahun;
}

File: Java/Java Kasus/src/com/myclass/Penjual.java
package com.myclass;

public class Penjual extends User {
    private int lamaKerjaPerTahun;

    public String displayUser() {
        return "ini Penjual";
    }

    public Penjual(String nama, int umur, String jenisKelamin, int lamaKerjaPerTahun) {
        super(nama, umur, jenisKelamin);
        this.lamaKerjaPerTahun = lamaKerjaPerTahun;
    }

    public void infoPenjual() {
        super.info();
        System.out.println(", Lama Kerja Per Tahun : " + this.lamaKerjaPerTahun + " Tahun.\n");
    }

    public void setLamaKerjaPerTahunPenjual(int lamaKerjaPerTahun) {
        this.lamaKerjaPerTahun = lamaKerjaPerTahun;
    }

    public int getLamaKerjaPerTahunPenjual() {
        return this.lamaKerjaPerTahun;
    }
}
